examples/vision/adamatch.py

```diff
@@ -0,0 +1,601 @@
+
+This scheduler increases the weight of the target domain loss from 0 to 1 for the first
+half of the training. Then it keeps that weight at 1 for the second half of the training.
+"""
+
+"""
+## Instantiate a Wide-ResNet-28-2
+
+The authors use a [WideResNet-28-2](https://arxiv.org/abs/1605.07146) for the dataset
+pairs we are using in this example. Most of the following code has been referred from
+[this script](https://github.com/asmith26/wide_resnets_keras/blob/master/main.py). Note
+that the following model has a scaling layer inside it that scales the pixel values to
+[0, 1].
+"""
+
+
+def wide_basic(n_input_plane, n_output_plane, stride):
+    def get_block(net):
+        conv_params = [[3, 3, stride, "same"], [3, 3, (1, 1), "same"]]
+
+        n_bottleneck_plane = n_output_plane
+
+        # Residual block
+        for i, v in enumerate(conv_params):
+            if i == 0:
+                if n_input_plane != n_output_plane:
+                    net = layers.BatchNormalization()(net)
+                    net = layers.Activation("relu")(net)
+                    convs = net
+                else:
+                    convs = layers.BatchNormalization()(net)
+                    convs = layers.Activation("relu")(convs)
+                convs = layers.Conv2D(
+                    n_bottleneck_plane,
+                    (v[0], v[1]),
+                    strides=v[2],
+                    padding=v[3],
+                    kernel_initializer=INIT,
+                    kernel_regularizer=regularizers.l2(WEIGHT_DECAY),
+                    use_bias=False,
+                )(convs)
+            else:
+                convs = layers.BatchNormalization()(convs)
+                convs = layers.Activation("relu")(convs)
+                convs = layers.Conv2D(
+                    n_bottleneck_plane,
+                    (v[0], v[1]),
+                    strides=v[2],
+                    padding=v[3],
+                    kernel_initializer=INIT,
+                    kernel_regularizer=regularizers.l2(WEIGHT_DECAY),
+                    use_bias=False,
+                )(convs)
+
+        # Shortcut connection: identity function or 1x1
+        # convolutional
+        #  (depends on difference between input & output shape - this
+        #   corresponds to whether we are using the first block in
+        #   each
+        #   group; see `layer()`).
+        if n_input_plane != n_output_plane:
+            shortcut = layers.Conv2D(
+                n_output_plane,
+                (1, 1),
+                strides=stride,
+                padding="same",
+                kernel_initializer=INIT,
+                kernel_regularizer=regularizers.l2(WEIGHT_DECAY),
+                use_bias=False,
+            )(net)
+        else:
+            shortcut = net
+
+        return layers.Add()([convs, shortcut])
+
+    return f
+
+
+# Stacking residual Units on the same stage
+def layer(block, n_input_plane, n_output_plane, count, stride):
```

Comment:
Just noticing this now, but that's *a lot* of nesting and functional indirection, which is harmful to readability. Consider doing:

```python
def block(x, n_input_plane, n_output_plane, stride):
     # One resnet block
     x = layers.BatchNormalization()(x)
     x = ...
     return x

def series_of_blocks(x, count, n_input_plane, n_output_plane, stride):
    x = block(x, n_input_plane, n_output_plane, stride=stride)
    for i in range(2, int(count + 1)):
        x = block(x, n_input_plane, n_output_plane, stride=1)
    return x

```

Reply:
Made it simpler for readability.